### DISCUSSION TITLE
Problems with sheetId and sheetName in 1.0.0

### DISCUSSION THREAD

**Discussion by nathipg** [handsontable/hyperformula#784] [General]
Hello!

I updated my Hyperformula to 1.0.0 and I start to have a error related to sheetId:

![image](https://user-images.githubusercontent.com/14853904/127142369-ba13938b-d9cd-4ea0-bb45-89696042c81c.png)

I've defined my sheetId and sheetName like this:

```
formulas: {
    engine: HyperFormula,
    sheetId: this.data.activeTabId, // Number
    sheetName: this.data.activeTabName, // String
},
```

Even though I have set my sheetId, Hyperformula initialize it with the value of sheetName.
If I don't set the sheetName, sheetId is automatically set as "Sheet1" which is not a number.
If I set sheetName as a number, sheetId is ok, but then sheetName gives an error, because it needs to be a string.

Looking into Hyperformula code in devTools, I found this:

![image](https://user-images.githubusercontent.com/14853904/127143858-7d336abf-0f02-4712-a30b-31e616d64c84.png)

So it looks like Hyperformula actually is using sheetName as sheetId, even though it is not possible, because id needs to be a number and name needs to be a string.

If I remove Hyperformula, Handsontable is initiated without errors, so I think the problem is not there.

I'm not sure if I'm missing something, do I need to configure something else?

**EDIT 1**

Debuging I found this issue:

![image](https://user-images.githubusercontent.com/14853904/127148230-8ba00fff-67ae-4c69-8a1f-181edd8f8b42.png)

![image](https://user-images.githubusercontent.com/14853904/127148018-24fc4336-f653-4439-aab4-cb199c0e1ebd.png)

isItPossibleToReplaceSheetContent waits for sheetId but onAfterLoadData pass sheetName to it

**Comment by MartinDawson** [comment]
Handsontable does not support hyperformula 1.0.0 yet.

```js
  "optionalDependencies": {
    "hyperformula": "0.6.2"
  },
```

https://github.com/handsontable/handsontable/blob/master/package.json

You will need to downgrade to 0.6.* if you use handsontable with hyperformula.

Also, handsontable issues should go on the [handsontable repo](https://github.com/handsontable/handsontable) I think. This is just for hyperformula.

**Reply by nathipg** [reply]
I'm sorry, I thought it was a hyperformula problem

**Reply by nathipg** [reply]
> Handsontable does not support hyperformula 1.0.0 yet.
> 
> ```js
>   "optionalDependencies": {
>     "hyperformula": "0.6.2"
>   },
> ```
> 
> https://github.com/handsontable/handsontable/blob/master/package.json
> 
> You will need to downgrade to 0.6.* if you use handsontable with hyperformula.
> 
> Also, handsontable issues should go on the [handsontable repo](https://github.com/handsontable/handsontable) I think. This is just for hyperformula.

Sorry, I know is not related but do you know where to find docs from 0.6.2?
Docs were updated to 1.0 and I can't find a way to look at the old one

**Reply by MartinDawson** [reply]
That is an issue indeed. I would recommend you raise an issue here for that and ask them to add links to the old archived docs.

You will need to do this in the meantime:

- Clone [hyperformula 0.6](https://github.com/handsontable/hyperformula/tree/release/0.6.2) locally and run it. The docs are built as part of hyperformula. You can find the files in the code base as they are called *.md.

Alternatively you can just view each file in the /docs/guide/*.md directory of 0.6.* hyperformula codebase: https://github.com/handsontable/hyperformula/tree/release/0.6.2/docs

**Reply by nathipg** [reply]
Thank you!
